sdk/python/lib/pulumi/automation/_cmd.py

```diff
@@ -208,29 +210,37 @@ def run(
         cmd.extend(args)
 
         stdout_chunks: List[str] = []
+        stderr_chunks: List[str] = []
 
-        with tempfile.TemporaryFile() as stderr_file:
-            with subprocess.Popen(
-                cmd, stdout=subprocess.PIPE, stderr=stderr_file, cwd=cwd, env=env
-            ) as process:
-                assert process.stdout is not None
-                while True:
-                    output = process.stdout.readline().decode(encoding="utf-8")
-                    if output == "" and process.poll() is not None:
-                        break
-                    if output:
-                        text = output.rstrip()
-                        if on_output:
-                            on_output(text)
-                        stdout_chunks.append(text)
-
-                code = process.returncode
-
-            stderr_file.seek(0)
-            stderr_contents = stderr_file.read().decode("utf-8")
+        with subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=cwd, env=env
+        ) as process:
+            assert process.stdout is not None
+            assert process.stderr is not None
+
+            while True:
+                output = process.stdout.readline().decode(encoding="utf-8")
+                error = process.stderr.readline().decode(encoding="utf-8")
```

Comment:
I think this assumes you always have output on both. If there's nothing on stderr for example, we'll hang here.

Probably best to `select` on the files:

```python
while True:
    fds = [p.stdout.fileno(), p.stderr.fileno()]
    (readable, _, _) = select.select(fds, [], [])

    for fd in readable:
        if fd == p.stdout.fileno():
            output = p.stdout.readline()
            text = output.rstrip()
            if on_output:
                on_output(text)
                stdout_chunks.append(text)
        if fd == p.stderr.fileno():
            read = p.stderr.readline()
    
     if p.poll() is not None:
        break
```

Reply:
`p` = `process`